src/components/storage/index.ts

```diff
@@ -334,3 +352,128 @@ export class IpfsStorage extends Storage {
+  parseDecryptedStream(decryptedStream: Readable): Promise<S3Object> {
+    return new Promise((resolve, reject) => {
+      let data = ''
+      decryptedStream.on('data', (chunk) => {
+        data += chunk
+      })
+      decryptedStream.on('end', () => {
+        try {
+          const parsedData = JSON.parse(data)
+          resolve(parsedData)
+        } catch (error) {
+          reject(error)
+        }
+      })
+      decryptedStream.on('error', (error) => {
+        reject(error)
+      })
+    })
+  }
+
+  getDownloadUrl(): string {
+    const fileHash = this.getFile().hash
+    return fileHash
+  }
+
+  async getS3Object(): Promise<S3Object> {
+    const file = this.getFile()
+    const fileStream = Readable.from(file.hash)
+    const streamString = await streamToString(fileStream)
+    const encryptedData = ethers.getBytes(streamString)
+    const decryptedData = await decryptData(encryptedData, file.encryptMethod)
+    return JSON.parse(decryptedData.toString()) as S3Object
+  }
+
+  async fetchData(): Promise<any> {
+    const s3Obj = await this.getS3Object()
+    const spacesEndpoint = new AWS.Endpoint(s3Obj.endpoint)
+    const s3 = new AWS.S3({
+      endpoint: spacesEndpoint,
+      accessKeyId: s3Obj.accessKeyId,
+      secretAccessKey: s3Obj.secretAccessKey,
+      region: s3Obj.region
+    })
+
+    const params = {
+      Bucket: s3Obj.bucket,
+      Key: s3Obj.objectKey
+    }
+    try {
+      const data = await s3.getObject(params).promise()
+      console.log('Successfully retrieved object from S3')
+      return data
+    } catch (err) {
+      console.error('Error fetching object from S3:', err)
+    }
+  }
+
+  async fetchSpecificFileMetadata(): Promise<FileInfoResponse> {
+    const data = await this.fetchData()
+    const s3Obj = await this.getS3Object()
+    return {
+      valid: true,
+      contentLength: data.ContentLength,
+      contentType: data.ContentType,
+      name: s3Obj.objectKey,
+      type: 's3',
+      encryptedBy: this.getFile().encryptedBy,
+      encryptMethod: this.getFile().encryptMethod
+    }
+  }
+
+  async encryptDataContent(
+    data: S3Object,
+    encryptionType: EncryptMethod.AES | EncryptMethod.ECIES
+  ): Promise<String> {
+    const genericAssetData = Uint8Array.from(Buffer.from(JSON.stringify(data)))
+    const encryptedData = await encryptData(genericAssetData, encryptionType)
+    const encryptedMetaData = hexlify(encryptedData)
+    return encryptedMetaData
+  }
+
+  async decryptDataContent(
+    hash: String,
+    encryptionType: EncryptMethod.AES | EncryptMethod.ECIES
+  ): Promise<Buffer> {
+    try {
+      const fileStream = Readable.from(hash)
+      const streamString = await streamToString(fileStream)
+      const encryptedData = ethers.getBytes(streamString)
+      const data = await decryptData(encryptedData, encryptionType)
+      return data
+    } catch (err) {
+      console.error('Error fetching object from S3:', err)
+    }
+  }
+
+  async encryptContent(
```

Comment:
this is not needed, we already have this in parent class, we only need the streams

Reply:
Storage class require to implement some functions.. One is encryptContent, so i have to implement it in S3Storage becouse this class extends Storage

Reply:
I will extend Storage class to handle encrypt/decrypt for streams, so we don't need this in S3 Class